FIX: disable submit button after click (#1031)

ernieblues opened the pull request:
Prevent multiple submissions by disabling the 'Submit' button after the first click.

Close #1031

johnhenley commented:
@WillStrohl pls review. This PR relies or includes same as #1416.

johnhenley commented:
@ernieblues can you please remove the csproj changes and just leave the changes to SubmitForm.cs ? Thanks!

johnhenley commented:
> @ernieblues can you please remove the csproj changes and just leave the changes to SubmitForm.cs ? Thanks!
@ernieblues I updated the csproj to remove duplicated tags; however, I do see that visual studio adds back duplicates, so I'm not sure it's a permanent fix. But please re-sync your repo, remove any csproj changes you have on this branch, and try to re-build. Let me know if any issues...

johnhenley commented:
@ernieblues I am going to 

> @johnhenley
> 
> I synced `.csproj` with `upstream/dev`. Only the debounce fix in SubmitForm.cs is in this PR now.
> 
> Just a heads up: it currently doesn’t build.

@ernieblues I'm looking at the build issues, but in the meantime, asking @Timo-Breumelhof  to review the javascript change.

Timo-Breumelhof reviewed (commented):
@ernieblues thank you for your contribution.

I have a few worries.

1. As this module can be on a page with several other modules, I would prefer the variable to use a more unique name. I would suggest:

dcnSubmitted

2. the type='text/javascript' attribute is not needed any more and can be removed (looks more modern ;-)

3. A note: if, in the future, we would want to add debounce to other buttons too, I would prefer to add a function to our central js file and call that. 

A pure JS approach could be:

form.addEventListener('submit', function (e) {
  const btn = form.querySelector('button[type="submit"]');
  btn.disabled = true;
});

(for future reference)

ernieblues commented:
@Timo-Breumelhof

Thanks for the feedback. I've renamed the flag to dcnSubmitted and removed the type='text/javascript' attribute as suggested.

As for centralizing the debounce logic, unfortunately, a pure JS approach using `querySelector('button[type="submit"]')` doesn’t apply here. There’s no standard button element with a `type="submit"` to select — the submit control is a custom server-generated `<div>` with an `onclick` handler and a dynamic `id="btnSubmit"`.
```
<div onclick="amPostback();" class="amimagebutton" onmouseover="this.className='amimagebutton_over';" onmouseout="this.className='amimagebutton';" style="width:50px;height:50px;">
	<img src="/DesktopModules/ActiveForums/themes/community-default//images/save32.png" hspace="0" vspace="0" border="0" align="absmiddle"><br><span>Submit</span>
</div>
```
In PR #1027, I created a centralized JavaScript debounce method in afcommon.js that selects the `dcf-topic-link` and `dcf-forum-link` classes. In this case the `btnSubmit` id is server side.

Thanks for the approval, @johnhenley.
